pkg/k8shandler/clusterrestart.go

```diff
@@ -0,0 +1,552 @@
+package k8shandler
+
+import (
+	"fmt"
+
+	api "github.com/openshift/elasticsearch-operator/pkg/apis/logging/v1"
+	"github.com/openshift/elasticsearch-operator/pkg/elasticsearch"
+	"github.com/openshift/elasticsearch-operator/pkg/utils"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+)
+
+type ClusterRestart struct {
+	client           elasticsearch.Client
+	clusterName      string
+	clusterNamespace string
+	scheduledNodes   []NodeTypeInterface
+}
+
+type Restarter struct {
+	scheduledNodes   []NodeTypeInterface
+	clusterName      string
+	clusterNamespace string
+	clusterStatus    *api.ElasticsearchStatus
+	nodeStatus       *api.ElasticsearchNodeStatus
+
+	precheck func() error
+	prep     func() error
+	main     func() error
+	post     func() error
+	recovery func() error
```

Comment:
Reading through the test cases it not easy to remember what precheck, prep or post is. PreCheck is bound to cluster health or noop. Prep is bound to optional/required PrimariesAndFlush or noop. Post is always waitAllNodesRejoinAndSetAllShards. 

Precheck -> pre
Prep -> setup
main -> do/execute
post -> post
recovery -> recover/recoverData

IMHO, all proposed methods work better in combination with the struct name. WDYT?

Reply:
they aren't meant to be closely tied to anything, they could in theory have any function called at that point in time.